uniffi_bindgen/src/bindings/python/templates/ErrorTemplate.py

```diff
@@ -3,47 +3,42 @@
 # {{ type_name }}
 # We want to define each variant as a nested class that's also a subclass,
 # which is tricky in Python.  To accomplish this we're going to create each
-# class separated, then manually add the child classes to the base class's
+# class separately, then manually add the child classes to the base class's
 # __dict__.  All of this happens in dummy class to avoid polluting the module
 # namespace.
-class UniFFIExceptionTmpNamespace:
-    class {{ type_name }}(Exception):
-        pass
+class {{ type_name }}(Exception):
+    pass
+
+{{ type_name }}Temp = {{ type_name }}
+
+class {{ type_name }}:
     {% for variant in e.variants() %}
     {%- let variant_type_name = variant.name()|class_name %}
 
     {%- if e.is_flat() %}
-    class {{ variant_type_name }}({{ type_name }}):
-        def __str__(self):
-            return "{{ type_name }}.{{ variant_type_name }}({})".format(repr(super().__str__()))
+    class {{ variant_type_name }}({{ type_name }}Temp):
+        pass
     {%- else %}
-    class {{ variant_type_name }}({{ type_name }}):
+    class {{ variant_type_name }}({{ type_name }}Temp):
         def __init__(self{% for field in variant.fields() %}, {{ field.name()|var_name }}{% endfor %}):
             {%- if variant.has_fields() %}
+            super().__init__(", ".join([
+                {%- for field in variant.fields() %}
+                '{{ field.name()|var_name }}={!r}'.format({{ field.name()|var_name }}),
+                {%- endfor %}
+            ]))
             {%- for field in variant.fields() %}
             self.{{ field.name()|var_name }} = {{ field.name()|var_name }}
             {%- endfor %}
             {%- else %}
             pass
             {%- endif %}
-
-        def __str__(self):
-            {%- if variant.has_fields() %}
-            field_parts = [
-                {%- for field in variant.fields() %}
-                '{{ field.name()|var_name }}={!r}'.format(self.{{ field.name()|var_name }}),
-                {%- endfor %}
-            ]
-            return "{{ type_name }}.{{ variant_type_name }}({})".format(', '.join(field_parts))
-            {%- else %}
-            return "{{ type_name }}.{{ variant_type_name }}()"
-            {%- endif %}
     {%- endif %}
 
     {{ type_name }}.{{ variant_type_name }} = {{ variant_type_name }}
```

Comment:
I think this should assign to `{{ type_name}}Temp.{{ variant_type_name }}` (or whatever temp name we end up going with).  That way users can do use `FooError.Bar` to refer to the variant, which matches the Rust syntax.

...Or maybe we should just give up on this altogether.  It feels more Pythonic to make the variants top-level classes.  The one issue there is that 2 errors could have variants with the same name, but maybe #1426 could address that.

Reply:
> I think this should assign to `{{ type_name}}Temp.{{ variant_type_name }}` (or whatever temp name we end up going with). That way users can do use `FooError.Bar` to refer to the variant, which matches the Rust syntax.

Hmm, interesting, why did it work before? Apparently the new class's name isn't in scope when declaring the class (which is probably the reason for this whole construct), so this still worked.

I changed it anyway, for clarity.